### PR TITLE
Commerce 4845 base pricelist cant be deleted

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by marco-leo** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
@marco-leo I've just asked help to @riccardo-alberti this tests are failing because the change I made was to forbid the delete for base PL while the test are trying to delete those PL

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
@riccardo-alberti last three commits

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
Hi @riccardo-alberti I'm still having some problems with tests, can you help me please?

File: modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/helper/CommerceBasePriceListHelper.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ private void _deleteCatalogBaseCommercePriceList(
 					commerceCatalog.getGroupId(), type);
 
 		if (catalogBaseCommercePriceList != null) {
-			_commercePriceListLocalService.deleteCommercePriceList(
+			_commercePriceListLocalService.deleteForceCommercePriceList(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/helper/CommerceBasePriceListHelper.java:125]: @riccardo-alberti is this right? @commando18 fyi

Comment (riccardo-alberti) [reply modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/helper/CommerceBasePriceListHelper.java:125]: @andrea-ale-sbarra yes that's the case where we need to force delete the base pricelist

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/internal/helper/CommerceBasePriceListHelper.java:125]: thanks

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
rebased

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/service/impl/CommercePriceListLocalServiceImpl.java [hunk 2]
```diff
@@ -412,14 +412,10 @@ public void cleanPriceListCache(long companyId) {
 	@Indexable(type = IndexableType.DELETE)
 	@Override
 	@SystemEvent(type = SystemEventConstants.TYPE_DELETE)
-	public CommercePriceList deleteCommercePriceList(
-			CommercePriceList commercePriceList)
+	public CommercePriceList deleteForceCommercePriceList(
```
Comment (dsitu) [inline modules/apps/commerce/commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/service/impl/CommercePriceListLocalServiceImpl.java:455]: We should call it this
```suggestion
	public CommercePriceList forceDeleteCommercePriceList(
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by riccardo-alberti** [comment]
please refer to https://github.com/liferay-commerce/liferay-portal/pull/1041. Closing this. Thx

**Pull request closed without merging**